source/agora/test/ValidatorRecurringEnrollment.d

```diff
@@ -270,3 +270,62 @@ unittest
     assert(blocks[$ - 1].header.enrollments.length == 3);
     assert(blocks[$ - 2].header.enrollments.length == 3);
 }
+
+// Some nodes are interrupted during their validator cycles, they should
+// still manage to enroll when they are back online
+unittest
+{
+    import core.thread;
+
+    TestConf conf = {
+        quorum_threshold : 66
+    };
+    auto network = makeTestNetwork(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    // Get the genesis block, make sure it's the only block externalized
+    auto blocks = node_1.getBlocksFrom(0, 2);
+    assert(blocks.length == 1);
+
+    // Approach end of the cycle
+    network.generateBlocks(Height(GenesisValidatorCycle - 2));
+
+    Thread.sleep(1.seconds);
+    // Make 2 nodes sleep
+    nodes[$ - 1].ctrl.sleep(60.seconds, true);
+    nodes[$ - 2].ctrl.sleep(60.seconds, true);
+
+    network.generateBlocks(iota(GenesisValidators - 2),
+        Height(GenesisValidatorCycle - 1));
+
+    // Wake one up right before cycle ends
+    nodes[$ - 2].ctrl.sleep(0.seconds);
+    // Let it catch up
+    Thread.sleep(10.seconds);
```

Comment:
There seem to be a lot of long unconditional sleep calls. Since a recent change, all the tests are running serially rather than in parallel. So this will further slow down the tests.

Reply:
For now could we try to just reduce the time? 10 seconds seems a lot?

Reply:
Changed them with `expectBlock`